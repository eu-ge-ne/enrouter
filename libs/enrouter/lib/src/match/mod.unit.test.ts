import { describe, test, expect } from "vitest";
import * as regexparam from "regexparam";

import { matchRoutes } from "./mod.js";

import type { Route } from "#lib/route/mod.js";

describe("matchRoutes", () => {
  test("no matches", () => {
    const routes: Route = {
      path: "/abc",
      test: regexparam.parse("/abc", true),
      modules: [],
      loaded: false,
      elements: {},
    };

    expect(matchRoutes({ routes, location: "/x" })).toMatchSnapshot();
  });

  test("1 match", () => {
    const routes: Route = {
      path: "/",
      test: regexparam.parse("/", true),
      modules: [],
      loaded: false,
      elements: {},
    };

    expect(matchRoutes({ routes, location: "/" })).toMatchSnapshot();
  });

  test("2 matches", () => {
    const routes: Route = {
      path: "/",
      test: regexparam.parse("/", true),
      modules: [],
      loaded: false,
      elements: {},
      tree: [
        {
          path: "/abc",
          test: regexparam.parse("/abc", true),
          modules: [],
          loaded: false,
          elements: {},
        },
      ],
    };

    expect(matchRoutes({ routes, location: "/abc" })).toMatchSnapshot();
  });

  test("1 match with params", () => {
    const routes: Route = {
      path: "/[:id]",
      test: regexparam.parse("/:id", true),
      modules: [],
      loaded: false,
      elements: {},
    };

    expect(matchRoutes({ routes, location: "/100" })).toMatchSnapshot();
  });
});
