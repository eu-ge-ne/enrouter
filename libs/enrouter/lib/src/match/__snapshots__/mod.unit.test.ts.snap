// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matchRoutes > 1 match 1`] = `
[
  {
    "isFull": true,
    "last": [Circular],
    "location": "/",
    "next": undefined,
    "params": {},
    "route": {
      "elements": {},
      "loaded": false,
      "modules": [],
      "path": "/",
      "test": {
        "keys": [],
        "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
      },
    },
  },
]
`;

exports[`matchRoutes > 1 match with params 1`] = `
[
  {
    "isFull": true,
    "last": [Circular],
    "location": "/100",
    "next": undefined,
    "params": {
      "id": "100",
    },
    "route": {
      "elements": {},
      "loaded": false,
      "modules": [],
      "path": "/[:id]",
      "test": {
        "keys": [
          "id",
        ],
        "pattern": /\\^\\\\/\\(\\[\\^/\\]\\+\\?\\)\\(\\?=\\$\\|\\\\/\\)/i,
      },
    },
  },
]
`;

exports[`matchRoutes > 2 matches 1`] = `
[
  {
    "isFull": false,
    "last": {
      "isFull": true,
      "last": [Circular],
      "location": "/abc",
      "next": undefined,
      "params": {},
      "route": {
        "elements": {},
        "loaded": false,
        "modules": [],
        "path": "/abc",
        "test": {
          "keys": [],
          "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
        },
      },
    },
    "location": "/",
    "next": {
      "isFull": true,
      "last": [Circular],
      "location": "/abc",
      "next": undefined,
      "params": {},
      "route": {
        "elements": {},
        "loaded": false,
        "modules": [],
        "path": "/abc",
        "test": {
          "keys": [],
          "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
        },
      },
    },
    "params": {},
    "route": {
      "elements": {},
      "loaded": false,
      "modules": [],
      "path": "/",
      "test": {
        "keys": [],
        "pattern": /\\^\\(\\?=\\$\\|\\\\/\\)/i,
      },
      "tree": [
        {
          "elements": {},
          "loaded": false,
          "modules": [],
          "path": "/abc",
          "test": {
            "keys": [],
            "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
          },
        },
      ],
    },
  },
  {
    "isFull": true,
    "last": [Circular],
    "location": "/abc",
    "next": undefined,
    "params": {},
    "route": {
      "elements": {},
      "loaded": false,
      "modules": [],
      "path": "/abc",
      "test": {
        "keys": [],
        "pattern": /\\^\\\\/abc\\(\\?=\\$\\|\\\\/\\)/i,
      },
    },
  },
]
`;

exports[`matchRoutes > no matches 1`] = `[]`;
